elmslie-core/src/main/java/vivid/money/elmslie/core/store/BindingContract.kt

```diff
@@ -0,0 +1,52 @@
+package vivid.money.elmslie.core.store
+
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+
+/**
+ * A contract for passing data between two stores
+ */
+class BindingContract<ParentEvent, ParentEffect, ParentState, ChildEvent, ChildEffect, ChildState>(
+    val parent: Store<ParentEvent, ParentEffect, ParentState>,
+    val child: Store<ChildEvent, ChildEffect, ChildState>,
```

Comment:
Since one of the issues addressed is an attempt to make "parent" and "child" more independent then maybe it's better to move away from this concepts in naming as well? 
Smth like `parent -> first, child -> second` (FirstEvent, SecondEvent etc.) seems a bit more appropriate here and in other files

Reply:
I don't mind, let's see what @opengamer29 has to say

Reply:
Yes First/Second make more sense for me.